bu_passport/scripts/cfa_event_scraper.py

```diff
@@ -0,0 +1,319 @@
+            )
+            if raw_time.lower() == "all day":
+                return "12:00am", "11:59pm"
+            start_time, end_time = (time.strip() for time in raw_time.split("-"))
+            return start_time, end_time
+        except Exception as e:
+            print("parse_time:", e)
+            return None, None
+
+    def parse_daytime_range(
+        start_daytime: str, end_daytime: str
+    ) -> Tuple[datetime, datetime] | Tuple[None, None]:
+        boston_tz = pytz.timezone("America/New_York")
+        cur_time = datetime.now(boston_tz)
+        cur_month = cur_time.month
+        cur_year = cur_time.year
+
+        def parse_daytime(date_str: str) -> datetime | None:
+            try:
+                # Remove ordinal suffixes (e.g., '12th' -> '12')
+                cleaned_date_str = re.sub(r"(\d+)(st|nd|rd|th)", r"\1", date_str)
+                parsed_date = datetime.strptime(cleaned_date_str, "%A %b %d %I:%M%p")
+                return boston_tz.localize(parsed_date)
+            except Exception as e:
+                print("parse_daytime:", e)
+                return None
+
+        try:
+            start = parse_daytime(start_daytime)
+            end = parse_daytime(end_daytime)
+            if start is None or end is None:
+                print("empty daytime")
+                return start, end
+
+            # Adjust the year based on the current month
+            if start.month >= cur_month:
+                start = start.replace(year=cur_year)
+            else:
+                start = start.replace(year=cur_year + 1)
+
+            if end.month >= cur_month:
+                end = end.replace(year=cur_year)
+            else:
+                end = end.replace(year=cur_year + 1)
+
+            return start, end
+        except Exception as e:
+            print("parse_daytime_range:", e)
+            return None, None
+
+    # find date
+    event_date = parse_date(raw_when_span)
+    start_time, end_time = parse_time(raw_when_span)
+
+    start_daytime = f"{event_date} {start_time}"
+    end_daytime = f"{event_date} {end_time}"
+    return parse_daytime_range(start_daytime, end_daytime)
+
+
+def scrape_event_location(raw_event: str) -> str | None:
+    span: str = raw_event.find("span", class_="bulp-event-where")
+    if not span:
+        return None
+    else:
+        return span.text.strip()
+
+
+def scrape_event_detail_link(raw_event: str) -> Tuple[str, str] | Tuple[None, None]:
+    try:
+        span: str = raw_event.find("div", class_="bulp-event-buttons")
+        if not span:
+            return None, None
+        a_tag = span.find("a", class_="bulp-event-readmore")
+        href = a_tag["href"]
+
+        parsed_url = urlparse(href)
+        query_params = parse_qs(parsed_url.query)
+        eid = query_params.get("eid", [None])[0]
+        return f"https://www.bu.edu{href}", eid
+    except:
+        return None, None
+
+
+def scrape_event_detail_page(soup: BeautifulSoup):
```

Comment:
I just thought of one more tiny thing: the `_event_` naming in these functions seems a bit redundant. Since you are working with an event, this is not really adding any value, right?

E.g. `scrape_detail_link(cfa_event)` → It is kinda implicit that you're operating in an event, so that name is just unnecessarily long.